src/domain/storage/storage-bucket/storage.bucket.service.ts

```diff
@@ -260,23 +260,29 @@ export class StorageBucketService {
     }
   }
 
-  public async addDocumentToBucketOrFail(
-    storageBucketId: string,
+  public async addDocumentToStorageBucketOrFail(
+    storageBucket: IStorageBucket,
     document: IDocument
   ): Promise<IDocument> | never {
-    const storage = await this.getStorageBucketOrFail(storageBucketId);
-
-    this.validateMimeTypes(storage, document.mimeType);
-    this.validateSize(storage, document.size);
-    document.storageBucket = storage;
+    this.validateMimeTypes(storageBucket, document.mimeType);
+    this.validateSize(storageBucket, document.size);
+    document.storageBucket = storageBucket;
 
     this.logger.verbose?.(
-      `Added document '${document.externalID}' on storage bucket: ${storage.id}`,
+      `Added document '${document.externalID}' on storage bucket: ${storageBucket.id}`,
       LogContext.STORAGE_BUCKET
     );
     return this.documentService.save(document);
   }
 
+  public async addDocumentToStorageBucketByIdOrFail(
+    storageBucketId: string,
+    document: IDocument
+  ): Promise<IDocument> | never {
+    const storageBucket = await this.getStorageBucketOrFail(storageBucketId);
+    return this.addDocumentToStorageBucketOrFail(storageBucket, document);
```

Comment:
deliberately missing the await?

Reply:
yeah, this is already returning the promise that I need to return, so if anyone wants to await after this return is fine, no?